src/Curation/Curation.router.ts

```diff
@@ -239,12 +239,30 @@ export class CurationRouter extends Router {
       )
     }
 
-    return this.updateCuration(
-      itemId,
-      ethAddress,
-      curationJSON,
-      CurationType.ITEM
-    )
+    try {
+      return this.updateCuration(
+        itemId,
+        ethAddress,
+        curationJSON,
+        CurationType.ITEM
+      )
+    } catch (error) {
+      if (error instanceof NonExistentCollectionError) {
+        throw new HTTPError(
+          'Not found',
+          { id: error.id, ethAddress },
+          STATUS_CODES.notFound
+        )
+      } else if (error instanceof HTTPError) {
+        throw error
+      } else {
+        throw new HTTPError(
```

Comment:
Question! why is this the only place where we're doing this? meaning checking for HTTPErrors and throwing a new one otherwise.

It seems like the code in insertCuration should do the same ( `if (error instanceof HTTPError)` )? and the other catches might need to throw a new HTTPError instead of `throw error` ?

Reply:
You're right! moved it to `validateAccessToCuration` so the errors can get catched every time.